src/appleseed-max-impl/appleseedinteractive/appleseedinteractive.h

```diff
@@ -97,14 +97,21 @@ class AppleseedInteractiveRender
     virtual void AbortRender() override;
 #endif
 
-    void update_camera_parameters(INode* camera);
-    void update_camera_transform(INode* camera);
+    void update_camera_object(INode* camera);
+    void update_render_view();
     InteractiveSession* get_render_session();
+    static VOID CALLBACK timer_proc(
+        _In_ HWND     hwnd,
+        _In_ UINT     uMsg,
+        _In_ UINT_PTR idEvent,
+        _In_ DWORD    dwTime
+    );
 
   private:
+    static AppleseedInteractiveRender*              m_current_interactive;
```

Comment:
This is essentially a global variable. Since it appears to be accessed by multiple threads, it needs to be guarded (by a lock, typically) whenever it's read or modified.

Reply:
Added global lock the same way as it's done in logtarget.cpp

Reply:
And moved `m_current_interactive` to be really global variable, not static member: `g_current_interactive`